### PR TITLE
Updating Notices

### PR DESCRIPTION
**Pull request by davidwessman**
## Notices: reworking
- Adding admin namespace
- Adding markdown for content
- Adding controller spec
- Addding CarrierWave for image upload
- Updating views

## Other
- Updating event and work_post to use new CarrierWave setup

### CODE REVIEW

**Comment by davidwessman** [comment]
@navro @jforberg 
Any feedback here? 😄

File: app/controllers/admin/notices_controller.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    notice = Notice.find(params[:id])
+
+    notice.destroy!
+    redirect_to admin_notices_url, notice: alert_destroy(Notice)
+  end
+
+  private
+
+  def authorize
+    can?(:modify, Notice)
```
Comment (jforberg) [inline app/controllers/admin/notices_controller.rb]: Is this correct? So you need modify permission also to read the notices? And is not your load_permissions_and_authorize_resource taking care of this at the top?

File: app/models/notice.rb [hunk 2]
```diff
@@ -1,64 +1,24 @@
...
-    if d_publish.nil?
-      self.d_publish = Time.zone.today
-    end
-    if d_remove.nil?
-      self.d_remove = '2094-03-25'
-    end
+  def published?
+    d_publish < Time.zone.now &&
+      (d_remove.nil? || d_remove > Time.zone.now)
   end
```
Comment (jforberg) [inline app/models/notice.rb]: Nice, getting rid of a lot of code.

**Comment by jforberg** [comment]
The authorise check in your controller looks wrong to me. Cancan is supposed to do the check automatically for standard controller actions, that's what the load_and_authorize_resource is for. Also no need to redefine can(action, object) as cancan already includes authorize!(action, object) which does the right thing.

Otherwise very good work. -1

File: app/controllers/admin/notices_controller.rb (hunk 1 above)
Comment (jforberg) [inline app/controllers/admin/notices_controller.rb]: Check out the standard cancan authorize! method which already does this.

Comment (davidwessman) [inline app/controllers/admin/notices_controller.rb]: Hm, the problem with Cancancan and the admin namespace is that it cannot tell the admin namespace controller apart form the normal controller.
In this specific case I could remove the NoticesController completely, hence I might not need to use the before action.

**Comment by davidwessman** [comment]
Since the `NoticesController` is completely removed I do not need the `before_action`.
But if there is one `show`-action for users and one for admins Cancancan cannot tell them apart.

**Comment by davidwessman** [comment]
@jforberg Looking okay now? 😄

**Comment by navro** [comment]
> @navro @jforberg
> Any feedback here? :smile:

Looks good to me!

**Pull request merged**